src/redux/events/eventsMiddleware.ts

```diff
@@ -42,6 +44,12 @@ export const setupEventsSubscription = createAsyncThunk(
         case 'ActionRemoved':
           thunkAPI.dispatch(removeAction(data.Uri));
           break;
+        case 'MissionAdded':
+          thunkAPI.dispatch(refreshMissions());
+          break;
+        case 'MissionRemoved':
+          thunkAPI.dispatch(refreshMissions());
+          break;
```

Comment:
These two cases could be one

```suggestion
case: "MissionAdded":
case: "MissionRemoved":
...
```